packages/core/admin/admin/src/pages/Home/HomePage.tsx

```diff
@@ -65,33 +64,28 @@ const UnstableHomePageCe = () => {
   const user = useAuth('HomePageCE', (state) => state.user);
   const displayName = user?.firstname ?? user?.username ?? user?.email;
   const getAllWidgets = useStrapiApp('UnstableHomepageCe', (state) => state.widgets.getAll);
-
-  const [loadedWidgets, setLoadedWidgets] = React.useState<
-    (Omit<ReturnType<typeof getAllWidgets>[number], 'component'> & {
-      component: React.ComponentType;
-    })[]
-  >([]);
+  const [loadedComponents, setLoadedComponents] = React.useState<{
+    [key: string]: React.ComponentType;
+  }>({});
+  const widgets = getAllWidgets();
 
   React.useEffect(() => {
     const loadComponents = async () => {
-      const widgets = getAllWidgets();
-
-      const loaded = await Promise.all(
-        widgets.map(async (widget) => {
-          const LazyComponent = await widget.component();
-
-          return {
-            ...widget,
-            component: LazyComponent,
-          };
-        })
+      const resolvedComponents = await Promise.all(widgets.map((widget) => widget.component()));
```

Comment:
This is gonna load them all at once right? I don't think it's ideal, it will be costly if one widget is longer to load

I'd recommend to create a new component for each widget instance and to manage the loading individually inside. It should also be simpler, no reduce and store needed

Reply:
100000000%